cmd/storage-class_test.go

```diff
@@ -0,0 +1,280 @@
+		expectedData   int
+		expectedParity int
+	}{
+		{1, reducedRedundancyStorageClass, xl.storageDisks, 14, 2},
+		{2, standardStorageClass, xl.storageDisks, 8, 8},
+		{3, "", xl.storageDisks, 8, 8},
+		{4, reducedRedundancyStorageClass, xl.storageDisks, 9, 7},
+		{5, standardStorageClass, xl.storageDisks, 10, 6},
+	}
+	for _, tt := range tests {
+		// Set env var for test case 4
+		if tt.name == 4 {
+			globalRRStorageClass.Parity = 7
+		}
+		// Set env var for test case 5
+		if tt.name == 5 {
+			globalStandardStorageClass.Parity = 6
+		}
+		data, parity := getDrivesCount(tt.sc, tt.disks)
+		if data != tt.expectedData {
+			t.Errorf("Test %d, Expected data disks %d, got %d", tt.name, tt.expectedData, data)
+			return
+		}
+		if parity != tt.expectedParity {
+			t.Errorf("Test %d, Expected parity disks %d, got %d", tt.name, tt.expectedParity, parity)
+			return
+		}
+	}
+}
+
+func TestObjectQuorumFromMeta(t *testing.T) {
+	ExecObjectLayerTestWithDirs(t, testObjectQuorumFromMeta)
+}
+
+func testObjectQuorumFromMeta(obj ObjectLayer, instanceType string, dirs []string, t TestErrHandler) {
+	// Reset global storage class flags
+	resetGlobalStorageEnvs()
+	bucket := getRandomBucketName()
+	object := "object"
+
+	// make data with more than one part
+	partCount := 3
+	data := bytes.Repeat([]byte("a"), int(globalPutPartSize)*partCount)
+	xl := obj.(*xlObjects)
+	xlDisks := xl.storageDisks
+
+	err := obj.MakeBucketWithLocation(bucket, globalMinioDefaultRegion)
+	if err != nil {
+		t.Fatalf("Failed to make a bucket %v", err)
+	}
+
+	// Object for test case 1
+	_, err = obj.PutObject(bucket, object, mustGetHashReader(t, bytes.NewReader(data), int64(len(data)), "", ""), nil)
+	if err != nil {
+		t.Fatalf("Failed to putObject %v", err)
+	}
+
+	parts, errs := readAllXLMetadata(xlDisks, bucket, object)
+
+	// Object for test case 2
+	object1 := "object1"
+	metadata1 := make(map[string]string)
+	metadata1["x-amz-storage-class"] = reducedRedundancyStorageClass
+	_, err = obj.PutObject(bucket, object1, mustGetHashReader(t, bytes.NewReader(data), int64(len(data)), "", ""), metadata1)
+	if err != nil {
+		t.Fatalf("Failed to putObject %v", err)
+	}
+
+	parts1, errs1 := readAllXLMetadata(xlDisks, bucket, object1)
+
+	// Object for test case 3
+	object2 := "object2"
+	metadata2 := make(map[string]string)
+	metadata2["x-amz-storage-class"] = standardStorageClass
+	_, err = obj.PutObject(bucket, object2, mustGetHashReader(t, bytes.NewReader(data), int64(len(data)), "", ""), metadata2)
+	if err != nil {
+		t.Fatalf("Failed to putObject %v", err)
+	}
+
+	parts2, errs2 := readAllXLMetadata(xlDisks, bucket, object2)
+
+	tests := []struct {
+		name                int
+		xl                  xlObjects
+		parts               []xlMetaV1
+		errs                []error
+		expectedReadQuorum  int
+		expectedWriteQuorum int
+		expectedError       error
+	}{
+		{1, *xl, parts, errs, 8, 9, nil},
+		{2, *xl, parts1, errs1, 14, 15, nil},
+		{3, *xl, parts2, errs2, 8, 9, nil},
```

Comment:
For this test, use different value for standardStorageClass? (now you can not differentiate if it is actually different from standard case).
Ie if set to 6, then outcome should be 10 and 11 ?